wunderground_upload.yaml

```diff
@@ -190,23 +204,23 @@ variables:
     {{ sensor_humidity if sensor_humidity == '' else (states(sensor_humidity) | float(default=0)) }}
   windspeedmph: >
     {% set unit_of_measurement = state_attr(sensor_windspeed, 'unit_of_measurement') %}
-    {{ sensor_windspeed if sensor_windspeed == '' or unit_of_measurement != 'km/h'
+    {{ (states(sensor_windspeed) | float(default=0)) if sensor_windspeed == '' or unit_of_measurement != 'km/h'
         else ((states(sensor_windspeed) | float(default=0)) * 0.6213) }}
   windgustmph: >
     {% set unit_of_measurement = state_attr(sensor_windgust, 'unit_of_measurement') %}
-    {{ sensor_windgust if sensor_windgust == '' or unit_of_measurement != 'km/h'
+    {{ (states(sensor_windgust)|float(default=0)) if sensor_windgust == '' or unit_of_measurement != 'km/h'
         else ((states(sensor_windgust) | float(default=0)) * 0.6213) }}
   winddir: >
     {{ sensor_winddir if sensor_winddir == ''
         else (states(sensor_winddir) | float(default=0)) }}
   rainin: >
     {% set unit_of_measurement = state_attr(sensor_rain, 'unit_of_measurement') %}
-    {{ sensor_rain if sensor_rain == ''
-        else ((states(sensor_rain) | float(default=0)) / 25.4 if unit_of_measurement == 'mm' else 1) }}
+    {{ (states(sensor_rain)|float(default=0)) if sensor_rain == '' or unit_of_measurement == 'in'
+        else ((states(sensor_rain) | float(default=0)) / 25.4 if unit_of_measurement == 'mm' ) }}
```

Comment:
This works, but seems nothing better than before

Reply:
This results in a value of 0 if the sensor is not provided, which you don't want to upload. My code was wrong to have a value of 1 when the sensor is provided. This is fixed in commit 9f637e2.